src/Reflection/ParametersAcceptor.php

```diff
@@ -19,6 +20,8 @@ public function getTemplateTypeMap(): TemplateTypeMap;
 
 	public function getResolvedTemplateTypeMap(): TemplateTypeMap;
 
+	public function getCallSiteVarianceMap(): TemplateTypeVarianceMap;
```

Comment:
We can safely add this method only on `ParametersAcceptorWithPhpDocs`. The name of that class should really be `ExtendedParametersAcceptor`, the current name is a legacy burden :)

In most places in PHPStan, you can actually rely on `ParametersAcceptorWithPhpDocs` being there.

`ParametersAcceptorWithPhpDocs` follows `ExtendedMethodReflection` - an interface you can call, but cannot implement because a method might be added on it at any time.

Reply:
So try to move this to `ParametersAcceptorWithPhpDocs`, I think not a lot of places will break.

Reply:
This is a list of `@api` interfaces where we can add methods: https://github.com/phpstan/phpstan-src/blob/1.11.x/src/Rules/Api/BcUncoveredInterface.php

Reply:
Noted, I'll take a look